widgetssdk/src/main/java/com/glia/widgets/engagement/completion/EngagementCompletionActivityWatcher.kt

```diff
@@ -26,26 +26,28 @@ internal class EngagementCompletionActivityWatcher(controller: EngagementComplet
 
     private fun handleState(activityReference: WeakReference<Activity>, event: OneTimeEvent<EngagementCompletionState>) {
         val state = event.value
-        val activity = activityReference.get() ?: return
+        val activity = activityReference.get()
 
         when {
-            event.consumed || activity.isFinishing -> Logger.d(TAG, "skipping state: $state activity: $activity")
+            event.consumed -> Logger.d(TAG, "skipping.. event: ${event.value} is already consumed")
+            //finish activities event if app is in a background
```

Comment:
It took me a while to understand this comment. It's confusing that comment says "app is in background" but event name is "QueuingOrEngagementEnded". I think we need to explain a bit more.